state-sync/aptos-data-client/src/aptosnet/mod.rs

```diff
@@ -503,95 +537,163 @@ impl DataSummaryPoller {
-                    )
-                    .await
-                    .map(Response::into_payload);
-                drop(timer);
-
-                // Check the storage summary response
-                let storage_summary = match result {
-                    Ok(storage_summary) => storage_summary,
-                    Err(error) => {
-                        error!(
-                            (LogSchema::new(LogEntry::StorageSummaryResponse)
-                                .event(LogEvent::PeerPollingError)
-                                .message("Error encountered when polling peer!")
-                                .error(&error)
-                                .peer(&peer))
-                        );
-                        continue;
-                    }
-                };
+    /// Fetches the next prioritized peer to poll or None
+    /// if an error was encountered.
+    fn fetch_prioritized_peer(&self) -> Option<PeerNetworkId> {
+        self.data_client
+            .fetch_prioritized_peer_to_poll()
+            .unwrap_or_else(|error| {
+                log_poller_error(error);
+                None
+            })
+    }
 
-                // Update the global storage summary and the summary for the peer
-                self.data_client.update_summary(peer, storage_summary);
-                self.data_client.update_global_summary_cache();
+    /// Fetches the next regular peer to poll based on the sample frequency
+    fn fetch_regular_peer(&self, always_poll: bool) -> Option<PeerNetworkId> {
+        if always_poll {
+            self.try_fetch_regular_peer()
+        } else {
+            sample!(SampleRate::Frequency(REGULAR_PEER_SAMPLE_FREQ), {
+                return self.try_fetch_regular_peer();
+            });
+            None
+        }
+    }
 
-                // Log the new global data summary and update the metrics
-                sample!(
-                    SampleRate::Duration(Duration::from_secs(GLOBAL_DATA_LOG_FREQ_SECS)),
-                    info!(
-                        (LogSchema::new(LogEntry::PeerStates)
-                            .event(LogEvent::AggregateSummary)
-                            .message(&format!(
-                                "Global data summary: {:?}",
-                                self.data_client.get_global_data_summary()
-                            )))
-                    );
-                );
-                sample!(
-                    SampleRate::Duration(Duration::from_secs(GLOBAL_DATA_METRIC_FREQ_SECS)),
-                    let global_data_summary = self.data_client.get_global_data_summary();
-                    update_advertised_data_metrics(global_data_summary);
+    /// Attempts to fetch the next regular peer to poll from the data client.
+    /// If an error is encountered, the error is logged and None is returned.
+    fn try_fetch_regular_peer(&self) -> Option<PeerNetworkId> {
```

Comment:
nit: there is fair bit of code duplication overall between functions for fetching priority and regular peer. One way  to reduce the duplication might be to modify these functions to take `is_priority` as a flag.

Reply:
Reduced the duplication. Note: I think the readability is a little reduced but it's not bad enough to disagree with the nit 😄